From dbaf3d9ce3bf3ccdfaff44a5f340b9980bd5052e Mon Sep 17 00:00:00 2001
From: Hans de Goede <hdegoede@redhat.com>
Date: Fri, 12 May 2017 14:37:55 +0200
Subject: [PATCH xkeyboard-config] Add evdev mappings for KEY_SOUND, KEY_UWB,
 KEY_WWAN and KEY_RFKILL

KEY_SOUND, KEY_WWAN and KEY_RFKILL are used on actual devices and
current xproto master defines keysyms for these. Add mappings for
these keys.

The XF86UWB keysm exists for quite a while now, as does the KEY_SOUND
evdev code for it, so lets adds a mapping for that too.

Note that this replaces the I255 to XF86Hibernate mapping in
xkb_symbols "evdev", I don't know where that came from, but it is
wrong, I255 was until now not defined in keycodes/evdev and
KEY_HIBERNATE is not defined in linux/input-event-codes.h .

BugLink: https://bugs.freedesktop.org/show_bug.cgi?id=100970
Cc: Bastien Nocera <bnocera@redhat.com>
Cc: Benjamin Berg <bberg@redhat.com>
Signed-off-by: Hans de Goede <hdegoede@redhat.com>
---
 keycodes/evdev | 2 ++
 symbols/inet   | 7 ++++---
 2 files changed, 6 insertions(+), 3 deletions(-)

diff --git a/keycodes/evdev b/keycodes/evdev
index 624ac68..51a6e26 100644
--- a/keycodes/evdev
+++ b/keycodes/evdev
@@ -281,6 +281,8 @@ default xkb_keycodes "evdev" {
 	<I251> = 251;	// #define KEY_BRIGHTNESS_CYCLE    243
 	<I252> = 252;	// #define KEY_BRIGHTNESS_ZERO     244
 	<I253> = 253;	// #define KEY_DISPLAY_OFF         245
+	<I254> = 254;	// #define KEY_WWAN                246
+	<I255> = 255;	// #define KEY_RFKILL              247
 
 	// Fake keycodes for virtual keys
 	<LVL3> =   92;
diff --git a/symbols/inet b/symbols/inet
index 5403e77..1e06aa9 100644
--- a/symbols/inet
+++ b/symbols/inet
@@ -184,7 +184,7 @@ xkb_symbols "evdev" {
     key <I218>   {      [ Print                 ]       };
 //  key <I219>   {      [ ]       }; // KEY_HP
     key <I220>   {      [ XF86WebCam            ]       };
-//  key <I221>   {      [ ]       }; // KEY_SOUND
+    key <I221>   {      [ XF86AudioPreset       ]       };
 //  key <I222>   {      [ ]       }; // KEY_QUESTION
     key <I223>   {      [ XF86Mail              ]       };
     key <I224>   {      [ XF86Messenger         ]       }; // KEY_CHAT
@@ -210,13 +210,14 @@ xkb_symbols "evdev" {
     key <I244>   {      [ XF86Battery           ]       };
     key <I245>   {      [ XF86Bluetooth         ]       };
     key <I246>   {      [ XF86WLAN              ]       };
+    key <I247>   {      [ XF86UWB               ]       };
 //  key <I249>   {      [ ]       }; // KEY_VIDEO_NEXT  --  drive next video source
 //  key <I250>   {      [ ]       }; // KEY_VIDEO_PREV  --  drive previous video source
 //  key <I251>   {      [ ]       }; // KEY_BRIGHTNESS_CYCLE  --  bright up, max++ == min
 //  key <I252>   {      [ ]       }; // KEY_BRIGHTNESS_ZERO  --  brightness off
 //  key <I253>   {      [ ]       }; // KEY_DISPLAY_OFF  --  turn off display
-//  key <I254>   {      [ ]       }; // KEY_WIMAX
-    key <I255>   {      [ XF86Hibernate         ]       }; // KEY_HIBERNATE
+    key <I254>   {      [ XF86WWAN              ]       }; // KEY_WWAN
+    key <I255>   {      [ XF86RFKill            ]       }; // KEY_RFKILL
 
     key <FK13>   {      [ XF86Tools         ]       };
     key <FK14>   {      [ XF86Launch5       ]       };
-- 
2.12.2

